controllers/pipeline/pipeline_adapter_test.go

```diff
@@ -825,5 +864,94 @@ var _ = Describe("Pipeline Adapter", Ordered, func() {
 				"skipping creation of a new Snapshot  PipelineRun.Namespace default PipelineRun.Name pipelinerun-build-sample Component.Name component-sample"
 			Expect(buf.String()).Should(ContainSubstring(expectedLogEntry))
 		})
+
+		It("ensures ephemeral environment is deleted for the given pipelineRun ", func() {
+			deploymentTarget = &applicationapiv1alpha1.DeploymentTarget{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "dt" + "-",
+					Namespace:    "default",
+				},
+				Spec: applicationapiv1alpha1.DeploymentTargetSpec{
+					DeploymentTargetClassName: "dtcls-name",
+					KubernetesClusterCredentials: applicationapiv1alpha1.DeploymentTargetKubernetesClusterCredentials{
+						DefaultNamespace:           "default",
+						APIURL:                     "https://url",
+						ClusterCredentialsSecret:   "secret-sample",
+						AllowInsecureSkipTLSVerify: false,
+					},
+				},
+			}
+			Expect(k8sClient.Create(ctx, deploymentTarget)).Should(Succeed())
+
+			deploymentTargetClaim = &applicationapiv1alpha1.DeploymentTargetClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "dtc" + "-",
+					Namespace:    "default",
+				},
+				Spec: applicationapiv1alpha1.DeploymentTargetClaimSpec{
+					DeploymentTargetClassName: applicationapiv1alpha1.DeploymentTargetClassName("dtcls-name"),
+					TargetName:                deploymentTarget.Name,
+				},
+			}
+			Expect(k8sClient.Create(ctx, deploymentTargetClaim)).Should(Succeed())
+
+			binding = &applicationapiv1alpha1.SnapshotEnvironmentBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "binding" + "-",
+					Namespace:    "default",
+				},
+				Spec: applicationapiv1alpha1.SnapshotEnvironmentBindingSpec{
+					Application: hasApp.Name,
+					Environment: hasEnv.Name,
+					Snapshot:    hasSnapshot.Name,
+					Components:  []applicationapiv1alpha1.BindingComponent{},
+				},
+			}
+			Expect(k8sClient.Create(ctx, binding)).Should(Succeed())
+
+			hasEnv.Spec.Configuration.Target = applicationapiv1alpha1.EnvironmentTarget{
+				DeploymentTargetClaim: applicationapiv1alpha1.DeploymentTargetClaimConfig{
+					ClaimName: deploymentTargetClaim.Name,
+				},
+			}
+			Expect(k8sClient.Update(ctx, hasEnv)).Should(Succeed())
+
+			adapter = NewAdapter(testpipelineRunComponent, hasComp, hasApp, ctrl.Log, k8sClient, ctx)
+			Expect(reflect.TypeOf(adapter)).To(Equal(reflect.TypeOf(&Adapter{})))
+
+			Eventually(func() bool {
+				result, err := adapter.EnsureEphemeralEnvironmentsCleanedUp()
+				return !result.CancelRequest && err == nil
+			}, time.Second*20).Should(BeTrue())
+
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, types.NamespacedName{
```

Comment:
While this was the way this unit test was set up in the PoC, in retrospect this looks like a fragile way to check if the environment was deleted. 
Maybe getting all environments (like it's done e.g. [here](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/snapshot/snapshot_adapter.go#L430-L434)) and checking that `len(environmentList.Items) == 0` would be better here?

Reply:
Alternatively, we could use this approach of [checking for apierr.IsNotFound](https://github.com/redhat-appstudio/managed-gitops/blob/main/appstudio-controller/controllers/appstudio.redhat.com/deploymenttargetbinder_controller_test.go#L111-L112) used in managed-gitops unit tests

Reply:
Yes, thanks very much!

Reply:
Updated.

Reply:
Thanks!